drivers/adc/adc_stm32.c

```diff
@@ -539,13 +553,15 @@ static int adc_stm32_init(struct device *dev)
 	 * These two series STM32 has one internal voltage reference source
 	 * to be enabled.
 	 */
-	LL_ADC_SetCommonPathInternalCh(ADC, LL_ADC_PATH_INTERNAL_VREFINT);
+	LL_ADC_SetCommonPathInternalCh(__LL_ADC_COMMON_INSTANCE(adc),
```

Comment:
Same here, please submit at least in a different commit explaining this change.

Reply:
Pulled this one out into its own #if defined. I'm not sure if it's MCU specific, but I wasn't able to compile with the original code.

Reply:
I see, the problem is I was using the legacy define of common adc instance pointer 'ADC', which should be deprecated and already commented as legacy in some SoCs' header files like stm32l083xx.h:782. There's no need to add another #if here, just change the 'ADC' in line 558 to __LL_ADC_COMMON_INSTANCE(adc) should be fine for all SoCs.